AEPAnalytics/Sources/Analytics.swift

```diff
@@ -36,16 +38,142 @@ public class Analytics: NSObject, Extension {
+        // If state is present then pageName = state name else pageName = app id to prevent hit from being discarded.
+        if let actionName = eventData[AnalyticsConstants.EventDataKeys.TRACK_ACTION] as? String, !actionName.isEmpty {
+            ret[AnalyticsConstants.AnalyticsRequestKeys.IGNORE_PAGE_NAME] =  AnalyticsConstants.IGNORE_PAGE_NAME_VALUE
+            let isInternal = eventData[AnalyticsConstants.EventDataKeys.TRACK_INTERNAL] as? Bool ?? false
+            ret[AnalyticsConstants.AnalyticsRequestKeys.ACTION_NAME] = getActionPrefix(isInternalAction: isInternal) + actionName
+        }
+        // Todo :- We currently read application id from lifecycle
+        //ret[AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME] = state->GetApplicationId();
+
+        if let stateName = eventData[AnalyticsConstants.EventDataKeys.TRACK_STATE] as? String, !stateName.isEmpty {
+            ret[AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME] = stateName
+        }
+
+        // Todo:- Aid. Should we add it to identity map or vars
+        // Todo:- Vid. Should we add it to identity map or vars
+
+        ret[AnalyticsConstants.AnalyticsRequestKeys.CHARSET] = AnalyticsConstants.CHARSET
+        ret[AnalyticsConstants.AnalyticsRequestKeys.FORMATTED_TIMESTAMP] = TimeZone.current.getOffsetFromGmtInMinutes()
+
+        // Set timestamp for all requests.
+        ret[AnalyticsConstants.AnalyticsRequestKeys.STRING_TIMESTAMP] = String(event.timestamp.getUnixTimeInSeconds())
+
+        // Todo:- GetAnalyticsIdVisitorParameters ??
+
+        if let appState = AnalyticsHelper.getApplicationState() {
+            ret[AnalyticsConstants.AnalyticsRequestKeys.CUSTOMER_PERSPECTIVE] = appState == .background ? AnalyticsConstants.APP_STATE_BACKGROUND : AnalyticsConstants.APP_STATE_FOREGROUND
+        }
+
+        return ret
+    }
+
+    private func processAnalyticsData(event: Event) -> [String: String] {
+        var ret = [String: String]()
+
+        // Todo:- Should we append default lifecycle context data (os version, device name, device version, etc) to each hits?
+
+        let contextData = event.data?[AnalyticsConstants.EventDataKeys.CONTEXT_DATA] as? [String: String] ?? [String: String]()
+        if !contextData.isEmpty {
+            contextData.forEach { (key, value) in ret[key] = value }
+        }
+
+        if let actionName = event.data?[AnalyticsConstants.EventDataKeys.TRACK_ACTION] as? String, !actionName.isEmpty {
+            let isInternal = event.data?[AnalyticsConstants.EventDataKeys.TRACK_INTERNAL] as? Bool ?? false
+            ret[getActionKey(isInternalAction: isInternal)] = actionName
+        }
+
+        // Todo :- Is TimeSinceLaunch" param is required? If so, calculate by listenining to lifecycle shared state update
+
+        if getPrivacyStatus(event: event) == .unknown {
+            ret[AnalyticsConstants.AnalyticsRequestKeys.PRIVACY_MODE] = "unknown"
```

Comment:
just curious, is there a reason the privacy status is added to the analytics data only if it is unknown?

Reply:
opt out drops the hit and opt in will send the hit.